### DISCUSSION TITLE
How 'fast-deep-equal'  needs to be added as a plugin in a Nuxt 3 implementation

### DISCUSSION THREAD

**Discussion by Stripkip** [NathanAP/vue-google-maps-community-fork#104] [Show and tell]
As mentioned in the README; if you don't include 'fast-deep-equal' to your project, you'll get this error:
`Requested module 'fast-deep-equal' does not provide an export named 'default'`

If you are using Nuxt 3 however, you'll need to fix this differently.

After you've added `fast-deep-equal` with `npm install fast-deep-equal`.
You'll need to configure `fast-deep-equal` as a plugin as well `~/plugin/fastDeepEqual.ts.`

```
import deepEqual from 'fast-deep-equal';

export default defineNuxtPlugin((nuxtApp) => {
    nuxtApp.provide('deepEqual', deepEqual);
});
```

(Maybe this should be added to the README [here](https://github.com/NathanAP/vue-google-maps-community-fork?tab=readme-ov-file#nuxt-3-usage).)

**Comment by vincentvdwal** [comment]
For me adding this to the `nuxt.config.ts` file worked aswell.

```
return defineNuxtConfig({
    ...
    vite: {
      optimizeDeps: {
        include: ['vue-google-maps-community-fork', 'fast-deep-equal']
      }
    }
    ...
})
